### PR TITLE
Api.py will warn (no more silence) on ImportError

### PR DESCRIPTION
**Pull request by maxkoryukov**
Hello!

I have added logging, when some modules are not installed. I have a pure _Linux Mint_ installation (pyyaml is not installed), and I have a project. There is `.coveralls.yml`, but `coveralls` does not work:

```bash
user@host ~/projects/project $ coveralls --verbose
You have to provide either repo_token in .coveralls.yml, or launch via Travis or CircleCI
```

I've spent a couple of hours, and found, that some exceptions were eaten in silence. It is bad. After patching:

```bash
user@host ~/projects/project $ coveralls --verbose
Seems, like some modules are not installed: No module named yaml
You have to provide either repo_token in .coveralls.yml, or launch via Travis or CircleCI
```

### Changes

`log.warn` added to exc.handler block. This code will just notify user on absence of some modules (in my case - yaml)

[Review on Reviewable](https://reviewable.io/reviews/coagulant/coveralls-python/104)

### CODE REVIEW

**Comment by maxkoryukov** [comment]
I didn't add any tests, but, hope , they are not necessary

**Comment by coagulant** [comment]
Makes sense to output only if yml file is present, otherwise it's not a warning, because yaml is optional and users should not worry about not having pyyaml.

**Comment by maxkoryukov** [comment]
I checked errors on Travis - they are related to the sending coverage report (for my second commit)

**Comment by maxkoryukov** [comment]
Oh... You are right about optional YAML. Let me add more checks

**Comment by maxkoryukov** [comment]
Now tests for new code are added, and check for access to file `.coveralls.yml` before `ImportError` assert is done

**Comment by maxkoryukov** [comment]
Seems, like this PR will close #99

File: tests/test_api.py [hunk 1]
```diff
@@ -59,6 +59,29 @@ def test_local_with_config(self):
...
+
+    @patch.object(log, 'warn')
+    def test_local_with_config_without_yaml_module(self, mock_logger):
+        """test local with config in yaml, but without yaml-installed"""
+        
+        yaml_import_mock = self.generate_import_mock('yaml', 'No module named yaml')
+        try:
+            with patch('__builtin__.__import__', side_effect=yaml_import_mock):
+                Coveralls()
+        except:
```
Comment (coagulant) [inline tests/test_api.py]: Why would you want to catch all exceptions, even those you're not expecting to get?

Comment (maxkoryukov) [inline tests/test_api.py]: I tried to cover just *necessary* behavior: warning on errors, during read config. Other tests are out of scope of this PR)) Sorry :-[

Comment (coagulant) [inline tests/test_api.py]: All I'm saying that catching all kinds of exceptions can hide a refactoring problem in future, that's all )
http://docs.quantifiedcode.com/python-anti-patterns/correctness/no_exception_type_specified.html

Anyway, this is not a PR merge blocker, just an advice

**Comment by coagulant** [comment]
Could you please rebase PR in single commit and add yuorself to AUTHORS, please?

File: tests/test_api.py (hunk 1 above)
Comment (maxkoryukov) [inline tests/test_api.py]: yeah, I understand.. but I (sadly) haven't enough time, to write super-hi-quality tests... Sorry my friend)))
Anyway, I got your advice, thanks)

**Comment by maxkoryukov** [comment]
Yes, wait plz

**Comment by maxkoryukov** [comment]
**Done**

**Comment by maxkoryukov** [comment]
Is there a chance that this PR will be merged ;)

**Comment by coagulant** [comment]
Sure, as soon Travis build is green on both py2.x 3.x )

**Comment by maxkoryukov** [comment]
The last commit has passed all tests for both 2.x and 3.x

**Comment by maxkoryukov** [comment]
How it is going??

**Pull request merged**